src/java.desktop/macosx/classes/sun/lwawt/macosx/CAccessibility.java

```diff
@@ -116,19 +120,14 @@ public void propertyChange(final PropertyChangeEvent evt) {
     private native void focusChanged();
 
     static <T> T invokeAndWait(final Callable<T> callable, final Component c) {
-        if (c != null) {
-            try {
-                return LWCToolkit.invokeAndWait(callable, c);
-            } catch (final Exception e) { e.printStackTrace(); }
-        }
-        return null;
+        return invokeAndWait(callable, c, null);
     }
 
     static <T> T invokeAndWait(final Callable<T> callable, final Component c, final T defValue) {
         T value = null;
         if (c != null) {
             try {
-                value = LWCToolkit.invokeAndWait(callable, c);
+                value = EventQueue.isDispatchThread() ? callable.call() : LWCToolkit.invokeAndWait(callable, c);
```

Comment:
When this method is called on EDT?

Reply:
In the same class there is a getChildrenAndRolesRecursive () method that calls invokeAndWait (), inside it it calls itself and getChildrenAndRoles (), which also call invokeAndWait ().

Reply:
ok, so this is in the new code, how hard it will be to reimplement these methods to call invokeAndWait only once per callback from the native, as it usually was done before?

Reply:
Done